scipio/src/sys/mod.rs

```diff
@@ -183,40 +183,46 @@ impl Wakers {
 }
 
 /// A registered source of I/O events.
-#[derive(Debug)]
+// #[derive(Debug)]
 pub struct InnerSource {
     /// Raw file descriptor on Unix platforms.
     pub(crate) raw: RawFd,
 
     /// Tasks interested in events on this source.
     pub(crate) wakers: RefCell<Wakers>,
 
-    pub(crate) source_type: SourceType,
+    pub(crate) source_type: RefCell<SourceType>,
 
     io_requirements: IoRequirements,
 
-    id: Option<SourceId>,
+    id: Cell<Option<SourceId>>,
+
+    queue: Cell<Option<ReactorQueue>>,
```

Comment:
@glommer this is a fun one -- you can use `Cell` with non-Copy types, it's a useful advanced tool to have in a toolbox.

Reply:
which is surprising, especially given that `Option` is not `Clone`